packages/shelter/src/flux.ts

```diff
@@ -113,3 +115,54 @@ export const storesFlat = new Proxy<Record<string, FluxStore>>(stores as any, {
     throw new Error("do not try to mutate flatStores");
   },
 });
+
+const storeInitPromises = new WeakMap<FluxStore, Promise<void>>();
+
+// awaits until the _isInitialized property of a store is true by overwritting it's setter
+function awaitStoreInit(store: FluxStore): Promise<void> {
+  if (storeInitPromises.has(store)) {
+    return storeInitPromises.get(store);
+  }
+  const initPromise = new Promise<void>((resolve) => {
+    if (store._isInitialized) return resolve();
+
+    let actualIsInitialized = false;
+    Object.defineProperty(store, "_isInitialized", {
+      get() {
+        return actualIsInitialized;
+      },
+      set(value) {
+        actualIsInitialized = value;
+        if (value === true) {
+          resolve();
```

Comment:
It would be nice to change this back to a standard property, instead of a custom one, once this is done, but it's not necessary to.

Reply:
Tought about this as well but the only way I know of would be to delete the property and redefine it. So I think I'm just gonna leave it be.

Reply:
That's true that's true, it's fine I guess!